addon/components/imgix-image.js

```diff
@@ -192,50 +159,130 @@ export default Component.extend({
         );
       }
 
+      // Setup
+      const widthProp = get(this, 'width');
+      const heightProp = get(this, 'height');
+      const pathAsUri = get(this, '_pathAsUri');
+      const disableSrcSet = get(this, 'disableSrcSet');
+      const client = get(this, '_client');
+      const buildWithOptions = options =>
+        client.buildURL(pathAsUri.path(), options);
+
+      const isFixedDimensionsMode = widthProp != null || heightProp != null;
+
+      const shouldShowDebugParams = get(config, 'APP.imgix.debug');
+
+      const { imgixOptions, aspectRatio } = (() => {
+        let imgixOptions = {
+          ...get(this, 'options')
+        };
+        let aspectRatio = get(this, 'aspectRatio');
+        if (imgixOptions.ar) {
+          aspectRatio = imgixOptions.ar;
+          delete imgixOptions.ar;
+        }
+
+        return { imgixOptions, aspectRatio };
+      })();
+
+      // Calculate AR
+      const aspectRatioDecimal = parseAspectRatio(aspectRatio);
+      if (aspectRatio != null && aspectRatioDecimal === false) {
+        // false return value from parseAspectRatio indicates invalid response
+        // eslint-disable-next-line no-console
+        console.warn(
+          `[imgix] The aspect ratio passed ("${aspectRatio}") is not in the correct format. The correct format is "W:H".`
+        );
+      }
+
+      // Ensure width and height are set correctly according to aspect ratio
+      const { width, height } = (() => {
+        if (widthProp && heightProp && aspectRatio) {
+          // eslint-disable-next-line no-console
+          console.warn(
+            `[imgix] All three of width, height, and aspect ratio were passed. The aspect ratio prop has not effect in this configuration.`
+          );
+        }
+        if (!widthProp && !heightProp && disableSrcSet && aspectRatio) {
+          // eslint-disable-next-line no-console
+          console.warn(
+            `[imgix] The aspect ratio prop has not effect when both width and height are not set, and when srcsets are disabled. To use aspect ratio, please either pass width or height values, or enable src sets.`
+          );
+        }
+
+        if (
+          !(widthProp || heightProp) ||
```

Comment:
Good grief, this conditional is a nightmare. I don't think there's an easy way to simplify it, though. Can we get a comment here, at least?